docs/branches.md

```diff
@@ -41,6 +40,59 @@ branch `main`. If the local target had also moved compared to `main@origin`
 merged. If one is ahead of the other, then that target will be the new target.
 Otherwise, the local branch will be conflicted (see next section for details).
 
+As of December 2023 Jujutsu tracks[^1] and fetches all branches by default, 
+which is confusing users coming from Git. To smoothen the transition branch 
+tracking was introduced. 
+
+### What does `git.auto-local-branches` actually do? 
+
+Jujutsu fetches all Git Refs under `origin/refs/branches` by default and then 
+creates local branches for them. This is similar to Mercurial, which fetches 
+all it's Booksmarks (equivalent to Git branches) by default. If you toggle the 
+feature off, Jujutsu will only create local branches for remote branches which 
+you explicitly track with `jj track`. 
+
+### Tracking a branch
+
+To track a branch permanently use `jj branch track <name-of-branch>@<remote>`. 
+It will now be imported as a local branch until you untrack it or it is deleted
+on the remote. 
+
+Example:
+
+```sh
+$ # List all available branches, as we want our colleague's branch.
+$ jj branch list --all
+$ # Find the branch.
+$ # [...]
+$ # Actually track the branch.
+$ jj branch track <name>@<remote> # Example: jj branch track my-feature@origin
```

Comment:
This could also be consistent with the notation elsewhere, e.g. `<branch name>@<remote name>`.

Reply:
done